### ISSUE TITLE
Can't Decrypt with Oaep algorithm

### ISSUE THREAD

**Issue by bauland** [Yubico/Yubico.NET.SDK#33]
Hello,
I would like to decrypt some random byte (fixed little string for test purpose) using piv certificate. All is working good with Pcks algorithm but not with Oaep. What I'm missing (I'm using Wpf .Net framework project)?
```
private void GetCert(PivSession session)
        {
            RSAEncryptionPadding encryptPaddingScheme = lbPadding.SelectedItems[0] as RSAEncryptionPadding;
            if (encryptPaddingScheme == null) return;

            byte[] encryptedData;
            byte[] rawDecryptedData;
            var slot = PivSlot.Signing;
            string data = "Yes ! It's working...";
            var plainTextBytes = Encoding.ASCII.GetBytes(data);
            int rsaFormatOaep = 0;
            bool isValid;
            byte[] decryptedData;
            switch (encryptPaddingScheme.OaepHashAlgorithm.Name)
            {
                case "SHA1":
                    rsaFormatOaep = RsaFormat.Sha1;
                    break;
                case "SHA256":
                    rsaFormatOaep = RsaFormat.Sha256;
                    break;
                case "384": //Unsupported from TryParsePkcs1Oaep
                    rsaFormatOaep = RsaFormat.Sha384;
                    break;
                case "SHA512":
                    rsaFormatOaep = RsaFormat.Sha512;
                    break;
            }

            try
            {
                // Get public key from slot
                var pivMetadata = session.GetMetadata(slot);
                if (pivMetadata.Algorithm.IsRsa()) Debug.WriteLine("Algorithm: ok (RSA)"); else return;

                var rsaPublic = (PivRsaPublicKey)pivMetadata.PublicKey;

                var rsaParams = new RSAParameters
                {
                    Modulus = rsaPublic.Modulus.ToArray(),
                    Exponent = rsaPublic.PublicExponent.ToArray()
                };
                // Encrypt data
                using (var rsa = RSA.Create(rsaParams))
                {
                    encryptedData = rsa.Encrypt(plainTextBytes, encryptPaddingScheme); // Encrypt with padding
                }

                // Decrypt data

                rawDecryptedData = session.Decrypt(slot, encryptedData);

                // Now unpadding
                isValid = encryptPaddingScheme.Mode == RSAEncryptionPaddingMode.Pkcs1 ? RsaFormat.TryParsePkcs1Decrypt(rawDecryptedData, out decryptedData) : RsaFormat.TryParsePkcs1Oaep(rawDecryptedData, rsaFormatOaep, out decryptedData);
                if (isValid)
                {
                    MessageBox.Show("Decrypted Message: " + Encoding.ASCII.GetString(decryptedData));
                }
                else
                {
                    MessageBox.Show("Decrypted failed.");
                }

            }
            catch (Exception exception)
            {
                MessageBox.Show(exception.Message);
            }

        }

```

**Comment by GregDomzalski** [comment]
Hi @bauland 

Thanks for your detailed bug report! Now that the team is back from winter holidays, I will have someone look into this issue.

**Comment by GregDomzalski** [comment]
Hey @bauland - a quick update. The team found a subtle difference in behavior in the .NET crypto APIs between .NET Framework 4.x and .NET 5+

Thankfully it's a straightforward fix and we will be releasing it in the next update (due out in less than a week).

**Comment by bauland** [comment]
Thanks for the infos. Hope it will be released soon.

**Comment by GregDomzalski** [comment]
This issue has been addressed in the latest release (1.6.0) which has now been released. Please give that a try.

I'm resolving this issue - however if you continue to encounter this issue, please feel free to reactivate.
